app/controllers/shopping_list_controller.rb

```diff
@@ -0,0 +1,21 @@
+class ShoppingListController < ApplicationController
+  before_action :authenticate_user!
+  def index
+    user = current_user
+    @missing_foods = []
+    @total_price = 0
+
+    user.recipes.includes(recipe_foods: :food).each do |recipe|
+      recipe.recipe_foods.each do |recipe_food|
+        available_food = recipe_food.food
+        next unless available_food && available_food.quantity < recipe_food.quantity
+
+        missing_quantity = recipe_food.quantity - available_food.quantity
+        missing_food = MissingFoodDecorator.new(available_food, missing_quantity)
+        @missing_foods << missing_food
```

Comment:
### [OPTIONAL]

You were supposed to keep the controllers thin and keep only relevant logic inside the controllers. These kinds of logic like missing_quantity, and missing_food should be implemented in a model and not in a controller.

Please try to implement this thing in a model instead of a controller because it's not a right place to implement a logic. 👍